tests/integration-tests/tests/createami/test_createami.py

```diff
@@ -153,22 +145,53 @@ def test_createami_wrong_pcluster_version(
     # Retrieve an AMI without 'aws-parallelcluster-<version>' in its name.
     # Therefore, we can bypass the version check in CLI and test version check of .bootstrapped file in Cookbook.
     wrong_ami = pcluster_ami_without_standard_naming(wrong_version)
+
+    command = _compose_command(region, instance, os, request, vpc_stack, wrong_ami, cluster_config)
+    pcluster_createami_result = run_command(command, raise_on_error=False)
+
+    logging.info("Verifying errors in createami stdout and packer log")
+    stdout = pcluster_createami_result.stdout
+    packer_log_path = _get_packer_log_path(stdout)
+    with open(packer_log_path) as f:
+        packer_log = f.read()
+    logging.info(packer_log)
+    assert_that(packer_log).contains("RuntimeError")
+    assert_that(packer_log).matches(fr"AMI was created.+{wrong_version}.+is.+used.+{current_version}")
+    assert_that(stdout).contains("No custom AMI created")
+
+
+def _compose_command(region, instance, os, request, vpc_stack, base_ami, cluster_config):
     # Networking
     vpc_id = vpc_stack.cfn_outputs["VpcId"]
-    subnet_id = vpc_stack.cfn_outputs["PublicSubnetId"]
-    networking_args = f"--vpc-id {vpc_id} --subnet-id {subnet_id}"
+    networking_args = ["--vpc-id", vpc_id, "--subnet-id", vpc_stack.cfn_outputs["PublicSubnetId"]]
 
     # Custom Cookbook
     custom_cookbook = request.config.getoption("createami_custom_chef_cookbook")
-    custom_cookbook_args = "" if not custom_cookbook else f"-cc {custom_cookbook}"
+    custom_cookbook_args = [] if not custom_cookbook else ["-cc", custom_cookbook]
 
-    command = (
-        f"pcluster createami -ai {wrong_ami} -os {os} -r {region} -c {cluster_config.as_posix()} "
-        f"{custom_cookbook_args} -i {instance} {networking_args}"
+    return (
+        [
+            "pcluster",
+            "createami",
+            "-ai",
+            base_ami,
+            "-os",
+            os,
+            "-r",
+            region,
+            "-c",
+            cluster_config.as_posix(),
+            "-i",
+            instance,
+        ]
+        + custom_cookbook_args
+        + networking_args
     )
-    createami_process = pexpect.spawn(command, timeout=600)
-    logging.info("Verifying errors in stdout")
-    createami_process.expect("RuntimeError")
-    createami_process.expect(fr"AMI was created.+{wrong_version}.+is.+used.+{current_version}")
-    createami_process.expect("No custom AMI created")
-    createami_process.close()
+
+
+def _get_packer_log_path(createami_stdout):
+    packer_log_key = "Packer log: "
+    for s in createami_stdout.split("\n"):
```

Comment:
minor: let's avoid single character variables, let's use a more self explanatory name (row? log_line?) 

Then what does it happen if there are no "s" starting with `packer_log_key` ? 
You should have a return value in any case.

Reply:
Done